### DISCUSSION TITLE
Usando 'linter' para BEM. É possível?

### DISCUSSION THREAD

**Discussion by SrVieira** [frontendbr/forum#1236] [Perguntas e Respostas]
Fala pessoal, tudo certo?

Estou aqui trabalhando na estrutura de um boilerplate para meus futuros projetos. O mesmo esta praticamente nos finalmente, eu resolvi criar um boilerplate pois tenho particularidades na estrutura de meus projetos e nenhum dos existentes atendia a estas.

A minha dúvida é a seguinte, seria possível definir um 'linter' para os arquivos de estilo e visões utilizando como padrão o BEM? Validar se cada um dos arquivos segue a convenção?

Peço desculpas se existe um post sobre isto, mas não encontrei aqui no forum. Agradeço desde já <3

**Comment by felipefialho** [comment]
Fala @SrVieira, tudo bem?

Existem algumas alternativas, em geral os linters de Sass ou Stylus, já fornecem opções para lintar o BEM.

Mas também é possível usar libs separadas, de todas [essa é a que mais gosto](https://github.com/postcss/postcss-bem-linter) especialmente por ser em PostCSS e ser agnóstica quanto ao pre-processador (ou não) utilizado.

**Comment by ninetails** [comment]
O [Stylelint](https://stylelint.io/) funciona de forma bem semelhante ao eslint e possui um [plugin pra BEM](https://github.com/simonsmith/stylelint-selector-bem-pattern) que utiliza a postcss-bem-linter que o @LFeh mandou. Nunca utilizei stylelint com BEM mas ainda sim recomendo por já possuir outras regras, apesar de achar que ele mal funciona, o que mais utilizei foi o order (e queria muito que funcionasse com o plugin do styled-components).

**Comment by SrVieira** [comment]
@ninetails @LFeh obrigado pelas dicas pessoal, vou estudar qual das ferramentas sugeridas se adequa ao que estou precisando <3

**Comment by lucianobarauna** [comment]
Acho post-css bem louco. Mas a ideia de ter um linter para BEM achei bem legal.

**Comment by eduardojmatos** [comment]
O sass-lint tem uma config "BEM simples" pra isso:

https://github.com/getninjas/gaiden/blob/master/.sass-lint.yml#L73-L76
